app/models/manageiq/providers/kubernetes/container_manager/scanning/job.rb

```diff
@@ -113,6 +113,21 @@ def pod_wait
     end
   end
 
+  def verify_scanned_image_id
+    metadata = image_inspector_client.fetch_metadata
+    actual = metadata.Id
+    return nil if actual == options[:docker_image_id]
+
+    if metadata.RepoDigests
+      metadata.RepoDigests.each do |repo_digest|
+        return nil if repo_digest == options[:docker_image_id]
+      end
+    end
+
+    "cannot analyze image %s with id %s: detected id was %s" % [
+      options[:image_full_name], options[:docker_image_id][0..11], actual[0..11]]
```

Comment:
We wouldn't know which one "should" have not failed as all the checks have already failed. I could also add the detected RepoDigests
Some thing like this:
```
    msg = "cannot analyze image %s with id %s: detected ids were %s" % [
      options[:image_full_name], options[:docker_image_id][0..11], actual[0..11]]
    metadata.RepoDigests.each do |digest|
      msg << digest.split('@').last[0..11] + ", "
    end
    return msg
```
I am splitting on '@' because RepoDigests include the image-repository and name but this will be less interesting in this message in my opinion.